libs/cardano-ledger-core/cddl/Cardano/Ledger/HasCDDL.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableSuperClasses #-}
+
+module Cardano.Ledger.HasCDDL (
+  HasCDDL (..),
+  huddleRule,
+  huddleGroup,
+  huddleGRule,
+) where
+
+import Cardano.Ledger.Core (Era)
+import Codec.CBOR.Cuddle.Huddle
+import Data.Proxy (Proxy (..))
+import GHC.TypeLits (KnownSymbol, Symbol, symbolVal)
+
+class Era era => HasCDDL (name :: Symbol) era where
```

Comment:
Let's split this type class into three separate ones. This will avoid usage of `error` and make the approach fully type safe:

```haskell
class (KnownSymbol name, Era era) => HuddleRule (name :: Symbol) era where
  huddleRule :: Proxy era -> Rule

class (KnownSymbol name, Era era) => HuddleGRule (name :: Symbol) era where
  huddleGRule :: Proxy era -> GRuleDef

class (KnownSymbol name, Era era) => HuddleGroup (name :: Symbol) era where
  huddleGroup :: Proxy era -> Named Group
```
Also CDDL in the name of a class isn't as suitable as Huddle IMHO.

Reply:
Awesome! Done!